gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/types/GradleFix.java

```diff
@@ -83,32 +91,47 @@ public SuggestedFix render(GradleFixContext context, VisitorState state) {
             NonAbstractGradleType.maybeTurnClassAbstract(context.enclosingClass.tree(), state)
                     .ifPresent(fixBuilder::merge);
         }
+        return fixBuilder.build();
+    }
 
-        // Inject services
-        servicesRequired.stream()
-                .filter(service -> !alreadyInjected(service, context.enclosingClass.tree(), state))
-                .forEach(service -> {
-                    fixBuilder.addImport(service.fullyQualifiedName());
-                    fixBuilder.addImport("javax.inject.Inject");
-                    SuggestedFix serviceGetter = SuggestedFixes.addMembers(
-                            context.enclosingClass.tree(),
-                            state,
-                            AdditionPosition.FIRST,
-                            String.format(
-                                    "@Inject\nprotected abstract %s %s();", service.className(), service.getterName()));
-                    fixBuilder.merge(serviceGetter);
+    private String getServiceReceiver(GradleFixContext context, VisitorState state, Builder fixBuilder) {
+        if (serviceMaybe.isEmpty()) {
+            // This means the method is on DefaultTask (i.e. `this`). No receiver needed.
+            return "";
+        }
+        GradleService service = serviceMaybe.get();
+        return findServiceReceiver(service, context.enclosingClass.tree(), state)
```

Comment:
Can we reference the method instead of the field?

```suggestion
        return findServiceReceiver(service, context.enclosingClass().tree(), state)
```